// @coingecko/coingecko-typescript/resources/onchain/networks/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrendingPools = exports.Tokens = exports.Pools = exports.NewPools = exports.Networks = exports.Dexes = void 0;
var dexes_1 = require("./dexes.js");
Object.defineProperty(exports, "Dexes", { enumerable: true, get: function () { return dexes_1.Dexes; } });
var networks_1 = require("./networks.js");
Object.defineProperty(exports, "Networks", { enumerable: true, get: function () { return networks_1.Networks; } });
var new_pools_1 = require("./new-pools.js");
Object.defineProperty(exports, "NewPools", { enumerable: true, get: function () { return new_pools_1.NewPools; } });
var index_1 = require("./pools/index.js");
Object.defineProperty(exports, "Pools", { enumerable: true, get: function () { return index_1.Pools; } });
var index_2 = require("./tokens/index.js");
Object.defineProperty(exports, "Tokens", { enumerable: true, get: function () { return index_2.Tokens; } });
var trending_pools_1 = require("./trending-pools.js");
Object.defineProperty(exports, "TrendingPools", { enumerable: true, get: function () { return trending_pools_1.TrendingPools; } });
