// @coingecko/coingecko-typescript/resources/onchain/networks/networks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Networks = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const DexesAPI = tslib_1.__importStar(require("./dexes.js"));
const dexes_1 = require("./dexes.js");
const NewPoolsAPI = tslib_1.__importStar(require("./new-pools.js"));
const new_pools_1 = require("./new-pools.js");
const TrendingPoolsAPI = tslib_1.__importStar(require("./trending-pools.js"));
const trending_pools_1 = require("./trending-pools.js");
const PoolsAPI = tslib_1.__importStar(require("./pools/pools.js"));
const pools_1 = require("./pools/pools.js");
const TokensAPI = tslib_1.__importStar(require("./tokens/tokens.js"));
const tokens_1 = require("./tokens/tokens.js");
class Networks extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.newPools = new NewPoolsAPI.NewPools(this._client);
        this.trendingPools = new TrendingPoolsAPI.TrendingPools(this._client);
        this.dexes = new DexesAPI.Dexes(this._client);
        this.pools = new PoolsAPI.Pools(this._client);
        this.tokens = new TokensAPI.Tokens(this._client);
    }
    /**
     * This endpoint allows you to **query all the supported networks on
     * GeckoTerminal**
     *
     * @example
     * ```ts
     * const network = await client.onchain.networks.get();
     * ```
     */
    get(query = {}, options) {
        return this._client.get('/onchain/networks', { query, ...options });
    }
}
exports.Networks = Networks;
Networks.NewPools = new_pools_1.NewPools;
Networks.TrendingPools = trending_pools_1.TrendingPools;
Networks.Dexes = dexes_1.Dexes;
Networks.Pools = pools_1.Pools;
Networks.Tokens = tokens_1.Tokens;
