// @coingecko/coingecko-mcp/tools/onchain/categories/get-onchain-categories.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.categories',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/categories',
    operationId: 'categories-list',
};
exports.tool = {
    name: 'get_onchain_categories',
    description: 'This endpoint allows you to **query all the supported categories on GeckoTerminal**',
    inputSchema: {
        type: 'object',
        properties: {
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: `1`',
            },
            sort: {
                type: 'string',
                description: 'sort the categories by field <br> Default value: `h6_volume_percentage_desc`',
                enum: [
                    'h1_volume_percentage_desc',
                    'h6_volume_percentage_desc',
                    'h12_volume_percentage_desc',
                    'h24_tx_count_desc',
                    'h24_volume_usd_desc',
                    'fdv_usd_desc',
                    'reserve_in_usd_desc',
                ],
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.onchain.categories.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
