// @coingecko/coingecko-mcp/tools/onchain/networks/dexes/get-networks-onchain-dexes.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.dexes',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/dexes',
    operationId: 'dexes-list',
};
exports.tool = {
    name: 'get_networks_onchain_dexes',
    description: 'This endpoint allows you to **query all the supported decentralized exchanges (DEXs) based on the provided network on GeckoTerminal**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: 1',
            },
        },
    },
};
const handler = async (client, args) => {
    const { network, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.dexes.get(network, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
