// @coingecko/coingecko-mcp/tools/coins/market-chart/get-range-coins-market-chart.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'coins.market_chart',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/coins/{id}/market_chart/range',
    operationId: 'coins-id-market-chart-range',
};
exports.tool = {
    name: 'get_range_coins_market_chart',
    description: 'This endpoint allows you to **get the historical chart data of a coin within certain time range in UNIX along with price, market cap and 24hr volume based on particular coin ID**',
    inputSchema: {
        type: 'object',
        properties: {
            id: {
                type: 'string',
            },
            from: {
                type: 'number',
                description: 'starting date in UNIX timestamp ',
            },
            to: {
                type: 'number',
                description: 'ending date in UNIX timestamp',
            },
            vs_currency: {
                type: 'string',
                description: 'target currency of market data <br> *refers to [`/simple/supported_vs_currencies`](/reference/simple-supported-currencies).',
            },
            interval: {
                type: 'string',
                description: 'data interval, leave empty for auto granularity',
                enum: ['5m', 'hourly', 'daily'],
            },
            precision: {
                type: 'string',
                description: 'decimal place for currency price value',
                enum: [
                    'full',
                    '0',
                    '1',
                    '2',
                    '3',
                    '4',
                    '5',
                    '6',
                    '7',
                    '8',
                    '9',
                    '10',
                    '11',
                    '12',
                    '13',
                    '14',
                    '15',
                    '16',
                    '17',
                    '18',
                ],
            },
        },
    },
};
const handler = async (client, args) => {
    const { id, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.coins.marketChart.getRange(id, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
