// @coingecko/coingecko-typescript/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenLists = exports.Simple = exports.Search = exports.Ping = exports.Onchain = exports.NFTs = exports.Key = exports.Global = exports.Exchanges = exports.ExchangeRates = exports.Derivatives = exports.Companies = exports.Coins = exports.AssetPlatforms = void 0;
var asset_platforms_1 = require("./asset-platforms.js");
Object.defineProperty(exports, "AssetPlatforms", { enumerable: true, get: function () { return asset_platforms_1.AssetPlatforms; } });
var coins_1 = require("./coins/coins.js");
Object.defineProperty(exports, "Coins", { enumerable: true, get: function () { return coins_1.Coins; } });
var companies_1 = require("./companies/companies.js");
Object.defineProperty(exports, "Companies", { enumerable: true, get: function () { return companies_1.Companies; } });
var derivatives_1 = require("./derivatives/derivatives.js");
Object.defineProperty(exports, "Derivatives", { enumerable: true, get: function () { return derivatives_1.Derivatives; } });
var exchange_rates_1 = require("./exchange-rates.js");
Object.defineProperty(exports, "ExchangeRates", { enumerable: true, get: function () { return exchange_rates_1.ExchangeRates; } });
var exchanges_1 = require("./exchanges/exchanges.js");
Object.defineProperty(exports, "Exchanges", { enumerable: true, get: function () { return exchanges_1.Exchanges; } });
var global_1 = require("./global/global.js");
Object.defineProperty(exports, "Global", { enumerable: true, get: function () { return global_1.Global; } });
var key_1 = require("./key.js");
Object.defineProperty(exports, "Key", { enumerable: true, get: function () { return key_1.Key; } });
var nfts_1 = require("./nfts/nfts.js");
Object.defineProperty(exports, "NFTs", { enumerable: true, get: function () { return nfts_1.NFTs; } });
var onchain_1 = require("./onchain/onchain.js");
Object.defineProperty(exports, "Onchain", { enumerable: true, get: function () { return onchain_1.Onchain; } });
var ping_1 = require("./ping.js");
Object.defineProperty(exports, "Ping", { enumerable: true, get: function () { return ping_1.Ping; } });
var search_1 = require("./search/search.js");
Object.defineProperty(exports, "Search", { enumerable: true, get: function () { return search_1.Search; } });
var simple_1 = require("./simple/simple.js");
Object.defineProperty(exports, "Simple", { enumerable: true, get: function () { return simple_1.Simple; } });
var token_lists_1 = require("./token-lists.js");
Object.defineProperty(exports, "TokenLists", { enumerable: true, get: function () { return token_lists_1.TokenLists; } });
