// @coingecko/coingecko-mcp/tools/coins/contract/get-coins-contract.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'coins.contract',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/coins/{id}/contract/{contract_address}',
    operationId: 'coins-contract-address',
};
exports.tool = {
    name: 'get_coins_contract',
    description: 'This endpoint allows you to **query all the metadata (image, websites, socials, description, contract address, etc.) and market data (price, ATH, exchange tickers, etc.) of a coin from the CoinGecko coin page based on an asset platform and a particular token contract address**',
    inputSchema: {
        type: 'object',
        properties: {
            id: {
                type: 'string',
            },
            contract_address: {
                type: 'string',
            },
        },
    },
};
const handler = async (client, args) => {
    const { contract_address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.coins.contract.get(contract_address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
