// @coingecko/coingecko-mcp/tools/simple/supported-vs-currencies/get-simple-supported-vs-currencies.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'simple.supported_vs_currencies',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/simple/supported_vs_currencies',
    operationId: 'simple-supported-currencies',
};
exports.tool = {
    name: 'get_simple_supported_vs_currencies',
    description: 'This endpoint allows you to **query all the supported currencies on CoinGecko**',
    inputSchema: {
        type: 'object',
        properties: {},
    },
};
const handler = async (client, args) => {
    return (0, types_1.asTextContentResult)(await client.simple.supportedVsCurrencies.get());
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
