// @coingecko/coingecko-mcp/tools/search/get-search.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'search',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/search',
    operationId: 'search-data',
};
exports.tool = {
    name: 'get_search',
    description: 'This endpoint allows you to **search for coins, categories and markets listed on CoinGecko**',
    inputSchema: {
        type: 'object',
        properties: {
            query: {
                type: 'string',
                description: 'search query',
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.search.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
