// @coingecko/coingecko-mcp/tools/nfts/get-id-nfts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'nfts',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/nfts/{id}',
    operationId: 'nfts-id',
};
exports.tool = {
    name: 'get_id_nfts',
    description: 'This endpoint allows you to **query all the NFT data (name, floor price, 24hr volume ...) based on the NFT collection ID**',
    inputSchema: {
        type: 'object',
        properties: {
            id: {
                type: 'string',
            },
        },
    },
};
const handler = async (client, args) => {
    const { id, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.nfts.getID(id));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
