// @coingecko/coingecko-mcp/tools/nfts/get-list-nfts.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'nfts',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/nfts/list',
    operationId: 'nfts-list',
};
exports.tool = {
    name: 'get_list_nfts',
    description: 'This endpoint allows you to **query all supported NFTs with ID, contract address, name, asset platform ID and symbol on CoinGecko**',
    inputSchema: {
        type: 'object',
        properties: {
            order: {
                type: 'string',
                description: 'use this to sort the order of responses',
                enum: [
                    'h24_volume_usd_asc',
                    'h24_volume_usd_desc',
                    'h24_volume_native_asc',
                    'h24_volume_native_desc',
                    'floor_price_native_asc',
                    'floor_price_native_desc',
                    'market_cap_native_asc',
                    'market_cap_native_desc',
                    'market_cap_usd_asc',
                    'market_cap_usd_desc',
                ],
            },
            page: {
                type: 'number',
                description: 'page through results',
            },
            per_page: {
                type: 'number',
                description: 'total results per page <br> Valid values: 1...250',
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.nfts.getList(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
