// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/top-holders/get-tokens-networks-onchain-top-holders.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.tokens.top_holders',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{address}/top_holders',
    operationId: 'top-token-holders-token-address',
};
export const tool = {
    name: 'get_tokens_networks_onchain_top_holders',
    description: 'This endpoint allows you to **query top token holders based on the provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            address: {
                type: 'string',
            },
            holders: {
                type: 'string',
                description: 'number of top token holders to return, you may use any integer or `max` <br> Default value: 10',
            },
        },
    },
};
export const handler = async (client, args) => {
    const { address, ...body } = args;
    return asTextContentResult(await client.onchain.networks.tokens.topHolders.get(address, body));
};
export default { metadata, tool, handler };
