// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/trades/get-tokens-networks-onchain-trades.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.tokens.trades',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{token_address}/trades',
    operationId: 'token-trades-contract-address',
};
export const tool = {
    name: 'get_tokens_networks_onchain_trades',
    description: 'This endpoint allows you to **query the last 300 trades in the past 24 hours, across all pools, based on the provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            token_address: {
                type: 'string',
            },
            trade_volume_in_usd_greater_than: {
                type: 'number',
                description: 'filter trades by trade volume in USD greater than this value <br> Default value: 0',
            },
        },
    },
};
export const handler = async (client, args) => {
    const { token_address, ...body } = args;
    return asTextContentResult(await client.onchain.networks.tokens.trades.get(token_address, body));
};
export default { metadata, tool, handler };
