// @coingecko/coingecko-mcp/tools/onchain/pools/trending-search/get-pools-onchain-trending-search.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.pools.trending_search',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/pools/trending_search',
    operationId: 'trending-search-pools',
};
exports.tool = {
    name: 'get_pools_onchain_trending_search',
    description: 'This endpoint allows you to **query all the trending search pools across all networks on GeckoTerminal**',
    inputSchema: {
        type: 'object',
        properties: {
            include: {
                type: 'string',
                description: 'attributes to include, comma-separated if more than one to include <br> Available values: `base_token`, `quote_token`, `dex`, `network`',
            },
            pools: {
                type: 'integer',
                description: 'number of pools to return, maximum 10 <br> Default value: 4',
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.onchain.pools.trendingSearch.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
