// @coingecko/coingecko-mcp/tools/coins/list/get-coins-list.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'coins.list',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/coins/list',
    operationId: 'coins-list',
};
export const tool = {
    name: 'get_coins_list',
    description: 'This endpoint allows you to **query all the supported coins on CoinGecko with coins ID, name and symbol**',
    inputSchema: {
        type: 'object',
        properties: {
            include_platform: {
                type: 'boolean',
                description: "include platform and token's contract addresses, default: false",
            },
            status: {
                type: 'string',
                description: 'filter by status of coins, default: active',
                enum: ['active', 'inactive'],
            },
        },
    },
};
export const handler = async (client, args) => {
    const body = args;
    return asTextContentResult(await client.coins.list.get(body));
};
export default { metadata, tool, handler };
