// @coingecko/coingecko-mcp/tools/coins/list/get-coins-list.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'coins.list',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/coins/list',
    operationId: 'coins-list',
};
exports.tool = {
    name: 'get_coins_list',
    description: 'This endpoint allows you to **query all the supported coins on CoinGecko with coins ID, name and symbol**',
    inputSchema: {
        type: 'object',
        properties: {
            include_platform: {
                type: 'boolean',
                description: "include platform and token's contract addresses, default: false",
            },
            status: {
                type: 'string',
                description: 'filter by status of coins, default: active',
                enum: ['active', 'inactive'],
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.coins.list.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
