// @coingecko/coingecko-mcp/tools/nfts/market-chart/get-nfts-market-chart.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'nfts.market_chart',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/nfts/{id}/market_chart',
    operationId: 'nfts-id-market-chart',
};
exports.tool = {
    name: 'get_nfts_market_chart',
    description: 'This endpoint allows you **query historical market data of a NFT collection, including floor price, market cap, and 24hr volume, by number of days away from now**',
    inputSchema: {
        type: 'object',
        properties: {
            id: {
                type: 'string',
            },
            days: {
                type: 'string',
                description: 'data up to number of days <br> Valid values: any integer or max',
            },
        },
    },
};
const handler = async (client, args) => {
    const { id, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.nfts.marketChart.get(id, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
