// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/get-address-networks-onchain-tokens.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.tokens',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{address}',
    operationId: 'token-data-contract-address',
};
export const tool = {
    name: 'get_address_networks_onchain_tokens',
    description: 'This endpoint allows you to **query specific token data based on the provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            address: {
                type: 'string',
            },
            include: {
                type: 'string',
                description: 'attributes to include',
                enum: ['top_pools'],
            },
        },
    },
};
export const handler = async (client, args) => {
    const { address, ...body } = args;
    return asTextContentResult(await client.onchain.networks.tokens.getAddress(address, body));
};
export default { metadata, tool, handler };
