// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/get-address-networks-onchain-tokens.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.tokens',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{address}',
    operationId: 'token-data-contract-address',
};
exports.tool = {
    name: 'get_address_networks_onchain_tokens',
    description: 'This endpoint allows you to **query specific token data based on the provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            address: {
                type: 'string',
            },
            include: {
                type: 'string',
                description: 'attributes to include',
                enum: ['top_pools'],
            },
        },
    },
};
const handler = async (client, args) => {
    const { address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.tokens.getAddress(address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
