// @coingecko/coingecko-mcp/tools/onchain/networks/get-onchain-networks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks',
    operationId: 'networks-list',
};
exports.tool = {
    name: 'get_onchain_networks',
    description: 'This endpoint allows you to **query all the supported networks on GeckoTerminal**',
    inputSchema: {
        type: 'object',
        properties: {
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: 1',
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
