// @coingecko/coingecko-typescript/resources/onchain/onchain.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as CategoriesAPI from "./categories.mjs";
import { Categories, } from "./categories.mjs";
import * as NetworksAPI from "./networks/networks.mjs";
import { Networks } from "./networks/networks.mjs";
import * as PoolsAPI from "./pools/pools.mjs";
import { Pools } from "./pools/pools.mjs";
import * as SearchAPI from "./search/search.mjs";
import { Search } from "./search/search.mjs";
import * as SimpleAPI from "./simple/simple.mjs";
import { Simple } from "./simple/simple.mjs";
import * as TokensAPI from "./tokens/tokens.mjs";
import { Tokens } from "./tokens/tokens.mjs";
export class Onchain extends APIResource {
    constructor() {
        super(...arguments);
        this.categories = new CategoriesAPI.Categories(this._client);
        this.networks = new NetworksAPI.Networks(this._client);
        this.pools = new PoolsAPI.Pools(this._client);
        this.search = new SearchAPI.Search(this._client);
        this.simple = new SimpleAPI.Simple(this._client);
        this.tokens = new TokensAPI.Tokens(this._client);
    }
}
Onchain.Categories = Categories;
Onchain.Networks = Networks;
Onchain.Pools = Pools;
Onchain.Search = Search;
Onchain.Simple = Simple;
Onchain.Tokens = Tokens;
