// @coingecko/coingecko-typescript/resources/onchain/onchain.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Onchain = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const CategoriesAPI = tslib_1.__importStar(require("./categories.js"));
const categories_1 = require("./categories.js");
const NetworksAPI = tslib_1.__importStar(require("./networks/networks.js"));
const networks_1 = require("./networks/networks.js");
const PoolsAPI = tslib_1.__importStar(require("./pools/pools.js"));
const pools_1 = require("./pools/pools.js");
const SearchAPI = tslib_1.__importStar(require("./search/search.js"));
const search_1 = require("./search/search.js");
const SimpleAPI = tslib_1.__importStar(require("./simple/simple.js"));
const simple_1 = require("./simple/simple.js");
const TokensAPI = tslib_1.__importStar(require("./tokens/tokens.js"));
const tokens_1 = require("./tokens/tokens.js");
class Onchain extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.categories = new CategoriesAPI.Categories(this._client);
        this.networks = new NetworksAPI.Networks(this._client);
        this.pools = new PoolsAPI.Pools(this._client);
        this.search = new SearchAPI.Search(this._client);
        this.simple = new SimpleAPI.Simple(this._client);
        this.tokens = new TokensAPI.Tokens(this._client);
    }
}
exports.Onchain = Onchain;
Onchain.Categories = categories_1.Categories;
Onchain.Networks = networks_1.Networks;
Onchain.Pools = pools_1.Pools;
Onchain.Search = search_1.Search;
Onchain.Simple = simple_1.Simple;
Onchain.Tokens = tokens_1.Tokens;
