// @coingecko/coingecko-mcp/tools/onchain/networks/new-pools/get-networks-onchain-new-pools.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.new_pools',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/new_pools',
    operationId: 'latest-pools-list',
};
exports.tool = {
    name: 'get_networks_onchain_new_pools',
    description: 'This endpoint allows you to **query all the latest pools across all networks on GeckoTerminal**',
    inputSchema: {
        type: 'object',
        properties: {
            include: {
                type: 'string',
                description: 'attributes to include, comma-separated if more than one to include <br> Available values: `base_token`, `quote_token`, `dex`, `network`',
            },
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: 1',
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.newPools.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
