// @coingecko/coingecko-mcp/tools/onchain/networks/new-pools/get-network-networks-onchain-new-pools.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.new_pools',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/new_pools',
    operationId: 'latest-pools-network',
};
export const tool = {
    name: 'get_network_networks_onchain_new_pools',
    description: 'This endpoint allows you to **query all the latest pools based on provided network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            include: {
                type: 'string',
                description: 'attributes to include, comma-separated if more than one to include <br> Available values: `base_token`, `quote_token`, `dex`',
            },
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: 1',
            },
        },
    },
};
export const handler = async (client, args) => {
    const { network, ...body } = args;
    return asTextContentResult(await client.onchain.networks.newPools.getNetwork(network, body));
};
export default { metadata, tool, handler };
