// @coingecko/coingecko-mcp/tools/coins/history/get-coins-history.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'coins.history',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/coins/{id}/history',
    operationId: 'coins-id-history',
};
exports.tool = {
    name: 'get_coins_history',
    description: 'This endpoint allows you to **query the historical data (price, market cap, 24hrs volume, ...) at a given date for a coin based on a particular coin ID**',
    inputSchema: {
        type: 'object',
        properties: {
            id: {
                type: 'string',
            },
            date: {
                type: 'string',
                description: 'the date of data snapshot <br> Format: `dd-mm-yyyy`',
            },
            localization: {
                type: 'boolean',
                description: 'include all the localized languages in response, default: true',
            },
        },
    },
};
const handler = async (client, args) => {
    const { id, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.coins.history.get(id, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
