// @coingecko/coingecko-mcp/tools/onchain/networks/pools/info/get-pools-networks-onchain-info.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.pools.info',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/pools/{pool_address}/info',
    operationId: 'pool-token-info-contract-address',
};
exports.tool = {
    name: 'get_pools_networks_onchain_info',
    description: 'This endpoint allows you to **query pool metadata (base and quote token details, image, socials, websites, description, contract address, etc.) based on a provided pool contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            pool_address: {
                type: 'string',
            },
        },
    },
};
const handler = async (client, args) => {
    const { pool_address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.pools.info.get(pool_address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
