// @coingecko/coingecko-mcp/tools/onchain/networks/pools/info/get-pools-networks-onchain-info.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.pools.info',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/pools/{pool_address}/info',
    operationId: 'pool-token-info-contract-address',
};
export const tool = {
    name: 'get_pools_networks_onchain_info',
    description: 'This endpoint allows you to **query pool metadata (base and quote token details, image, socials, websites, description, contract address, etc.) based on a provided pool contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            pool_address: {
                type: 'string',
            },
        },
    },
};
export const handler = async (client, args) => {
    const { pool_address, ...body } = args;
    return asTextContentResult(await client.onchain.networks.pools.info.get(pool_address, body));
};
export default { metadata, tool, handler };
