// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/pools/get-tokens-networks-onchain-pools.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.tokens.pools',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{token_address}/pools',
    operationId: 'top-pools-contract-address',
};
export const tool = {
    name: 'get_tokens_networks_onchain_pools',
    description: 'This endpoint allows you to **query top pools based on the provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            token_address: {
                type: 'string',
            },
            include: {
                type: 'string',
                description: 'attributes to include, comma-separated if more than one to include <br> Available values: `base_token`, `quote_token`, `dex`',
            },
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: 1',
            },
            sort: {
                type: 'string',
                description: 'sort the pools by field <br> Default value: h24_volume_usd_liquidity_desc',
                enum: ['h24_volume_usd_liquidity_desc', 'h24_tx_count_desc', 'h24_volume_usd_desc'],
            },
        },
    },
};
export const handler = async (client, args) => {
    const { token_address, ...body } = args;
    return asTextContentResult(await client.onchain.networks.tokens.pools.get(token_address, body));
};
export default { metadata, tool, handler };
