// @coingecko/coingecko-mcp/tools/onchain/networks/pools/trades/get-pools-networks-onchain-trades.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.pools.trades',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/pools/{pool_address}/trades',
    operationId: 'pool-trades-contract-address',
};
exports.tool = {
    name: 'get_pools_networks_onchain_trades',
    description: 'This endpoint allows you to **query the last 300 trades in the past 24 hours based on the provided pool address**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            pool_address: {
                type: 'string',
            },
            token: {
                type: 'string',
                description: "return trades for token <br> use this to invert the chart <br> Available values: 'base', 'quote' or token address <br> Default value: 'base'",
            },
            trade_volume_in_usd_greater_than: {
                type: 'number',
                description: 'filter trades by trade volume in USD greater than this value <br> Default value: 0',
            },
        },
    },
};
const handler = async (client, args) => {
    const { pool_address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.pools.trades.get(pool_address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
