// @coingecko/coingecko-mcp/tools/onchain/networks/pools/get-address-networks-onchain-pools.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.pools',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/pools/{address}',
    operationId: 'pool-address',
};
exports.tool = {
    name: 'get_address_networks_onchain_pools',
    description: 'This endpoint allows you to **query the specific pool based on the provided network and pool address**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            address: {
                type: 'string',
            },
            include: {
                type: 'string',
                description: 'attributes to include, comma-separated if more than one to include <br> Available values: `base_token`, `quote_token`, `dex`',
            },
            include_volume_breakdown: {
                type: 'boolean',
                description: 'include volume breakdown, default: false',
            },
        },
    },
};
const handler = async (client, args) => {
    const { address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.pools.getAddress(address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
