// @coingecko/coingecko-mcp/tools/asset-platforms/get-asset-platforms.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'asset_platforms',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/asset_platforms',
    operationId: 'asset-platforms-list',
};
exports.tool = {
    name: 'get_asset_platforms',
    description: 'This endpoint allows you to **query all the asset platforms on CoinGecko**',
    inputSchema: {
        type: 'object',
        properties: {
            filter: {
                type: 'string',
                description: 'apply relevant filters to results',
                enum: ['nft'],
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.assetPlatforms.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
