// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/holders-chart/get-tokens-networks-onchain-holders-chart.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.tokens.holders_chart',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{token_address}/holders_chart',
    operationId: 'token-holders-chart-token-address',
};
exports.tool = {
    name: 'get_tokens_networks_onchain_holders_chart',
    description: 'This endpoint allows you to **get the historical token holders chart based on the provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            token_address: {
                type: 'string',
            },
            days: {
                type: 'string',
                description: 'number of days to return the historical token holders chart <br> Default value: 7',
                enum: ['7', '30', 'max'],
            },
        },
    },
};
const handler = async (client, args) => {
    const { token_address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.tokens.holdersChart.get(token_address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
