// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/info/get-tokens-networks-onchain-info.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.tokens.info',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{address}/info',
    operationId: 'token-info-contract-address',
};
exports.tool = {
    name: 'get_tokens_networks_onchain_info',
    description: 'This endpoint allows you to **query token metadata (name, symbol,  CoinGecko ID, image, socials, websites, description, etc.) based on a provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            address: {
                type: 'string',
            },
        },
    },
};
const handler = async (client, args) => {
    const { address, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.tokens.info.get(address, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
