// @coingecko/coingecko-mcp/tools/onchain/networks/tokens/info/get-tokens-networks-onchain-info.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { asTextContentResult } from '@coingecko/coingecko-mcp/tools/types';
export const metadata = {
    resource: 'onchain.networks.tokens.info',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/tokens/{address}/info',
    operationId: 'token-info-contract-address',
};
export const tool = {
    name: 'get_tokens_networks_onchain_info',
    description: 'This endpoint allows you to **query token metadata (name, symbol,  CoinGecko ID, image, socials, websites, description, etc.) based on a provided token contract address on a network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            address: {
                type: 'string',
            },
        },
    },
};
export const handler = async (client, args) => {
    const { address, ...body } = args;
    return asTextContentResult(await client.onchain.networks.tokens.info.get(address, body));
};
export default { metadata, tool, handler };
