// @coingecko/coingecko-mcp/tools/onchain/networks/trending-pools/get-network-networks-onchain-trending-pools.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'onchain.networks.trending_pools',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/onchain/networks/{network}/trending_pools',
    operationId: 'trending-pools-network',
};
exports.tool = {
    name: 'get_network_networks_onchain_trending_pools',
    description: 'This endpoint allows you to **query the trending pools based on the provided network**',
    inputSchema: {
        type: 'object',
        properties: {
            network: {
                type: 'string',
            },
            duration: {
                type: 'string',
                description: 'duration to sort trending list by <br> Default value: 24h',
                enum: ['5m', '1h', '6h', '24h'],
            },
            include: {
                type: 'string',
                description: 'attributes to include, comma-separated if more than one to include <br> Available values: `base_token`, `quote_token`, `dex`',
            },
            page: {
                type: 'integer',
                description: 'page through results <br> Default value: 1',
            },
        },
    },
};
const handler = async (client, args) => {
    const { network, ...body } = args;
    return (0, types_1.asTextContentResult)(await client.onchain.networks.trendingPools.getNetwork(network, body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
