// @coingecko/coingecko-mcp/tools/search/trending/get-search-trending.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.handler = exports.tool = exports.metadata = void 0;
const types_1 = require("@coingecko/coingecko-mcp/tools/types");
exports.metadata = {
    resource: 'search.trending',
    operation: 'read',
    tags: [],
    httpMethod: 'get',
    httpPath: '/search/trending',
    operationId: 'trending-search',
};
exports.tool = {
    name: 'get_search_trending',
    description: 'This endpoint allows you **query trending search coins, NFTs and categories on CoinGecko in the last 24 hours**',
    inputSchema: {
        type: 'object',
        properties: {
            show_max: {
                type: 'string',
                description: 'show max number of results available for the given type <br> Available values: `coins`, `nfts`, `categories` <br> Example: `coins` or `coins,nfts,categories`',
            },
        },
    },
};
const handler = async (client, args) => {
    const body = args;
    return (0, types_1.asTextContentResult)(await client.search.trending.get(body));
};
exports.handler = handler;
exports.default = { metadata: exports.metadata, tool: exports.tool, handler: exports.handler };
